ITR/data/base_providers.py

```diff
@@ -569,38 +633,66 @@ def _standardize(self, intensities: pd.DataFrame) -> pd.DataFrame:
+                else:
+                    nominal_intensities = historic_intensities.apply(lambda x: x.map(lambda y: y.m if isinstance(y, pint.Quantity) else np.nan))
+                # See https://github.com/hgrecco/pint-pandas/issues/114
+                winsorized: pd.DataFrame = nominal_intensities.clip(
+                    lower=nominal_intensities.quantile(q=self.projection_controls.LOWER_PERCENTILE, axis='index',
+                                                        numeric_only=False),
+                    upper=nominal_intensities.quantile(q=self.projection_controls.UPPER_PERCENTILE, axis='index',
+                                                        numeric_only=False),
+                    axis='columns'
+                )
+                if ITR.HAS_UNCERTAINTIES:
+                    if uncertain_intensities.values.sum() != 0:
+                        uwinsorized = winsorized.apply(lambda x: PA_(ITR.uarray(x.values.data, uncertain_intensities[x.name].values), dtype=units[x.name]))
+                        return uwinsorized
+                winsorized_and_unitized = winsorized.apply(lambda x: PA_(x.values.data, dtype=units[x.name]))
+                return winsorized_and_unitized
+            except AttributeError:
+                breakpoint()
 
     def _interpolate(self, historic_intensities: pd.DataFrame) -> pd.DataFrame:
         # Interpolate NaNs surrounded by values, and extrapolate NaNs with last known value
-        interpolated = historic_intensities.copy()
-        for col in interpolated.columns:
-            if interpolated[col].isnull().all():
-                continue
-            qty = interpolated[col].values.quantity
-            s = pd.Series(data=qty.m, index=interpolated.index)
-            interpolated[col] = pd.Series(PA_(s.interpolate(method='linear', inplace=False, limit_direction='forward'),
-                                              dtype=f"{qty.u:~P}"), index=interpolated.index)
-        return interpolated
+        df = historic_intensities.copy()
+        for col in df.columns:
+            pa = PA_._from_sequence(df[col])
+            try:
+                if ITR.isnan(pa.data).all():
+                    continue
+                # pd.Series.interpolate only works on numeric data, so push down into PintArray
+                # FIXME: throw some uncertainty into the mix.  If we see ... X NA Y ... and interpolat the NA as Z as function of X and Y
+                ser = pd.Series(data=pa.data, index=df.index)
+                df[col] = pd.Series(PA_(ser.interpolate(method='linear', inplace=False, limit_direction='forward').values,
+                                        dtype=pa.units),
+                                    dtype=pa.dtype, index=df.index)
+            except AttributeError:
+                breakpoint()
+            except ValueError:
+                breakpoint()
+            except StopIteration:
+                breakpoint()
```

Comment:
do we want to mute these exceptions?

Reply:
More reminders that we need better code coverage testing and/or other test cases to ensure we're not past the end of our rope.

Reply:
Also removed...